Extension/src/LanguageServer/extension.ts

```diff
@@ -477,83 +515,79 @@ async function installVsix(vsixLocation: string, updateChannel: string): Promise
+                        reject(new Error('Failed to download VSIX package with proxySupport off')); // Changing the proxySupport didn't help.
+                        return;
                     }
-                    break;
-                }
-                try {
-                    await installVsix(vsixPath, updateChannel);
-                } catch (error) {
-                    reject(error);
+                    if (config.get('http.proxySupport') !== "off" && originalProxySupport !== "off") {
+                        config.update('http.proxySupport', "off", true);
+                        continue;
+                    }
+                    reject(new Error('Failed to download VSIX package'));
                     return;
                 }
-                clearInterval(insiderUpdateTimer);
-                const message: string =
-                    `The C/C++ Extension has been updated to version ${buildInfo.name}. Please reload the window for the changes to take effect.`;
-                util.promptReloadWindow(message);
-                telemetry.logLanguageServerEvent('installVsix', { 'success': 'true' });
-
-                resolve();
-            });
-        }, (error: Error) => {
-            // Handle getTargetBuildInfo rejection
-            reject(error);
+                if (originalProxySupport !== config.inspect<string>('http.proxySupport').globalValue) {
+                    config.update('http.proxySupport', originalProxySupport, true); // Reset the http.proxySupport.
+                    telemetry.logLanguageServerEvent('installVsix', { 'error': "Success with proxySupport off", 'success': 'true' });
+                }
+                break;
+            }
+            try {
+                await installVsix(vsixPath, updateChannel);
+            } catch (error) {
+                reject(error);
+                return;
+            }
+            clearInterval(insiderUpdateTimer);
+            const message: string =
+                `The C/C++ Extension has been updated to version ${buildInfo.name}. Please reload the window for the changes to take effect.`;
+            util.promptReloadWindow(message);
+            telemetry.logLanguageServerEvent('installVsix', { 'success': 'true' });
+            resolve();
         });
     });
-    await p.catch((error: Error) => {
-        // Handle .then following getTargetBuildInfo rejection
+    return p;
+}
+
+/**
+ * Query package.json and the GitHub API to determine whether the user should update, if so then install the update.
+ * The update can be an upgrade or downgrade depending on the the updateChannel setting.
+ * @param updateChannel The user's updateChannel setting.
+ */
+async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
+    let buildInfo: BuildInfo;
+    try {
+        buildInfo = await getTargetBuildInfo(updateChannel);
+        if (!buildInfo) {
+            return undefined;
```

Comment:
we don't usually return 'undefined' for functions marked 'void'.

Reply:
Isn't it because it's a Promise<void>?

Reply:
It's an `async` function, so you don't return a `Promise` explicitly. You return the wrapped type: `void` in this case.

Reply:
Yes. I think we should change `return undefined` to `return`.